### PR TITLE
chore(button): update variants and colors to match UI kit

### PR DESCRIPTION
**Pull request by dierat**
### Summary:
This PR updates the `Button` variants to match our styles and also adds the `color` prop.

I ported over the styles in the way we're doing them currently in EDS, which is with local CSS variables. The reason why we're using local CSS variables is because of the intersection of these 4 variants and these 5 colors.

I'm open to all feedback but I'm especially interested in ideas around other ways to handle this styling (or, if we keep using this method, what should the local variables be called to differentiate them from the top level ones?). This structure is the main reason why I think we should keep the `ClickableStyle` component for attaching styles to buttons and links, but maybe there's a better way to do it that would allow us to use mixins or something?

There is also a little bit of weirdness in terms of which colors are used where and when. It's not super consistent (except for the `disabled` state where they all use neutral 300 and `white` or `transparent`), which can be hard to see when looking at the buttons laid out, so I made a little table. These represent the color levels used across 4 of the colors options (neutral is slightly lighter because it feels so heavy): https://docs.google.com/spreadsheets/d/1zSgRkM-Q1jWcEj98pZGLLUBq4HTW5p0Ev0eaqOFlgmI

EDS button variant color per state  table

You can see the full spread of all the variants x colors x sizes x states on the main storybook: https://chanzuckerberg.github.io/edu-design-system/?path=/story/button--all-variants

all button variants, sizes, and states in the main storybook

Please note I haven't really touched the `inverted` styles; I don't want to spend too much time on that feature since I don't know for sure if we're going to support it.

### Test Plan:
Manually tested in storybook.

### CODE REVIEW

File: src/components/Button/Button.module.css [hunk 1]
```diff
@@ -177,6 +196,114 @@
...
+    --primary-color: var(--eds-color-brand-blue-800);
+    --plain-background-color: var(--eds-color-brand-blue-600);
+  }
+}
+
+/**
+  * Brand button
+  */
+.button--color-alert {
+  --primary-color: var(--eds-color-other-red-600);
```
[the remark this answers is not in the document]
Comment (dierat) [reply src/components/Button/Button.module.css:227]: @bradfrost oo thanks for bringing up the color token usage. I'm actually kind of struggling to understand how the system is intended to be used.

I'll use `brand` as an example just so I can talk about specifics here. In `colors-brand.json`, the only tokens exposed are levels 300, 600, and 800. However, based on the current design of the buttons, we need 100, 600, 700, and 800. Should I make another `tier-2-usage` file and place all the button tokens I need in there, drawing from the `tier-1-definitions` files? (Similar to `colors-link.json`?)

[the remark this answers is not in the document]
Comment (dierat) [reply src/components/Button/Button.module.css:227]: I see the tokens readme mentions tier 3 for component variables, but I don't see that in the actual codebase.

[the remark this answers is not in the document]
Comment (dierat) [reply src/components/Button/Button.module.css:227]: Maybe something like this? https://github.com/chanzuckerberg/edu-design-system/compare/diedra/button/3_variants...diedra/button/button-tokens (I know it's not following the pattern of foreground, background, etc, but that would be a ton of variables to do that for ever state for every variant for every color, so I just grabbed the 4 tokens we need for each color.)

[the remark this answers is not in the document]
Comment (anniehu4) [reply src/components/Button/Button.module.css:227]: there was a [`tier-3-components` folder in the original filedump](https://github.com/chanzuckerberg/eds-big-medium/tree/95ec7449e07ee3f1cea0db8bfc710c2f3223e9cf/src/design-tokens/tier-3-components), but nothing was in there and so we removed it

that `3_variants` branch looks reasonable to me 🤔  maybe while Brad is out next week @ifrost1 or @tpitre-czi could weigh in?

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -68,7 +72,7 @@ export interface Props {
   /**
    * Available _stylistic_ variations available for the Button component
    */
-  variant?: 'primary' | 'bare' | 'link' | 'table-header';
+  variant?: 'flat' | 'outline' | 'link' | 'plain';
```
Comment (anniehu4) [inline src/components/Button/Button.tsx:75]: I also wonder ... if we can confirm with design on cutting down the styles, perhaps we could only have `variant` with no `color` prop, in which case defining tokens is less unwieldy. Do you have a sense of when that convo might be resolved?

**Review by anniehu4 — APPROVED** [review]

**Comment by dierat** [comment]
This work is continuing in https://github.com/chanzuckerberg/edu-design-system/pull/936

**Pull request closed without merging**